module/apmhttp/client.go

```diff
@@ -139,11 +139,11 @@ func (r *roundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
 func SetHeaders(req *http.Request, traceContext apm.TraceContext, propagateLegacyHeader bool) {
 	headerValue := FormatTraceparentHeader(traceContext)
 	if propagateLegacyHeader {
-		req.Header.Set(ElasticTraceparentHeader, headerValue)
+		req.Header[ElasticTraceparentHeader] = []string{headerValue}
```

Comment:
+1 on the change but out of curiosity, won't they have the same allocs? Probably a bit high CPU cost in case of using `Set` due to canonical conversion of header keys but allocation wise I think they would be same.